roles/os_hardening/tasks/hardening.yml

```
@@ -1,4 +1,10 @@
 ---
+- name: Gather facts, if not present
+  ansible.builtin.setup:
+  when: ansible_facts == {}
```

```suggestion
  when: not ansible_facts
```

It probably should be like this.

See this example play:

```
- hosts: localhost
  gather_facts: false
  tasks:
  - debug:
      var: ansible_facts
  - name: Gather facts, if not present
    ansible.builtin.setup:
    when: not ansible_facts
    tags:
      - always
  - debug:
      var: ansible_facts
```